src/Http/Client/Cookie/CookieCollection.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.5.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Http\Client\Cookie;
+
+class CookieCollection
+{
+    /**
+     * @var Cookie[]
+     */
+    protected $cookies = [];
+
+    /**
+     * Constructor
+     *
+     * @param array $cookies Array of cookie objects
+     */
+    public function __construct(array $cookies = [])
+    {
+        $this->checkCookies($cookies);
+        foreach ($cookies as $cookie) {
+            $name = $cookie->getName();
+            $key = mb_strtolower($name);
```

Comment:
http://stackoverflow.com/questions/11311893/is-the-name-of-a-cookie-case-sensitive